### DISCUSSION TITLE
Code sample - custom link dialog

### DISCUSSION THREAD

**Discussion by shaipetel** [mdx-editor/editor#408] [Q&A]
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] I have read the documentation and cannot find an answer.

**Describe the bug**
I would like to have a custom link dialog that will allow me to open my own picker. I could not find any working code to pass a LinkDialog JSX element to the linkDialogPlugin

**Expected behavior**
Any code sample with a demo dialog for editing and inserting links

**Additional context**
In my application, there are many link suggestions and they are too long so the user cannot read them in the OOB dialog.
I would like to open a page picker that will allow custom navigation to pick a page in a folder-view and pages.

**Comment by elfenheart** [comment]
I was having the same requirements, to modify the dialog to conform to some other design style guidelines. I was able to decipher the functionality required by referencing the original Link Dialog: https://github.com/mdx-editor/editor/blob/main/src/plugins/link-dialog/LinkDialog.tsx

I think the main functionality that the use of `useCellValues` hooks to get the existing value of the URL and the `usePublisher` for publishing update from the dialog.

It is a bit confusing, so perhaps a documentation of these methods would be very helpful

**Reply by shaipetel** [reply]
Thanks! I will give it a try.

**Reply by petyosi** [reply]
@elfenheart Here's a section in the docs: https://mdxeditor.dev/editor/docs/extending-the-editor. 
The hooks are documented in the state management library: https://mdx-editor.github.io/gurx/;
Each available cell is documented in the API reference section: [https://mdxeditor.dev/editor/api/variables/updateLink_](https://mdxeditor.dev/editor/api/variables/updateLink_)
